run_dir/static/js/data_del_plot.js

```diff
@@ -0,0 +1,357 @@
+                series_name = "Metagenomics";
+            }else if (data[d][1].application.includes('novo')){
+                series_name = "de novo";
+            }else if (data[d][1].application.includes('Epigenetics')){
+                series_name = "Epigenetics";
+            }else{
+                series_name = "Other/undefined";
+            }
+            view_color = view_coloring(series_name);
+        }else if (view_type == 'type'){
+            if (data[d][1].type == null){
+                series_name = "Other/undefined";
+            }else{
+                series_name = data[d][1].type;
+            }
+            view_color = view_coloring(series_name);
+        }else if (view_type == 'delivery_type'){
+            if (data[d][1].delivery_type == null){
+                series_name = "Other/undefined";
+            }else{
+                series_name = data[d][1].delivery_type;
+            }
+            view_color = view_coloring(series_name);
+        }else if (view_type == 'sample_type'){
+            if (data[d][1].sample_type.includes('Finished')){
+                series_name = "Finished Library";
+            }else if (data[d][1].sample_type.includes('total')){
+                series_name = "total RNA";
+            }else if (data[d][1].sample_type.includes('Tissue')){
+                series_name = "Tissue";
+            }else if (data[d][1].sample_type.includes('Genomic')){
+                series_name = "Genomic DNA";
+            }else if (data[d][1].sample_type.includes('Amplicon')){
+                series_name = "Amplicon";
+            }else{
+                series_name = "Other/undefined";
+            }
+            view_color = view_coloring(series_name);
+        }else if (view_type == 'best_practice_bioinformatics'){
+            if (data[d][1].best_practice_bioinformatics == 'Yes'){
+                series_name = "BP bioinformatics";
+            }else if (data[d][1].best_practice_bioinformatics == 'No'){
+                series_name = "No BP bioinformatics";
+            }else if (data[d][1].best_practice_bioinformatics == 'Special'){
+                series_name = "Special";
+            }else{
+                series_name = "Other/undefined";
+            }
+            view_color = view_coloring(series_name);
+            }
+        if (!series.hasOwnProperty(series_name)){
+            series[series_name] = {
+                name: series_name,
+                color: view_color,
+                data: [],
+            };
+            series.length += 1;
+        }
+        dp = {
+            y: data[d][1][key],
+            name: date_close,
+            pm: project_name,
+            ownURL: bioinfo_link
+        };
+        series[series_name].data.push(dp);
+        categories.push(date_close);
+        // Hackery to get a proper JS array for HCharts
+        var proper_series = []
+        for (s in series) {
+            proper_series.push(series[s]);
+        }
```

Comment:
I don't totally understand why you need this, but if you're going to do it then this is cleaner:

```suggestion
        var proper_series = Object.values(series)
```

(I _think_ it should be equivalent)

Reply:
I copied i from the flowcell trends plot :-) it doesn't work without it though, the person who originally wrote the code also commented that it's needed. But you are right, your code is cleaner!

Reply:
This kind of thing smells like it needs a bigger-picture code refactor. But if it works then I'm happy 😉